charts/posthog/templates/_kafka.tpl

```diff
@@ -0,0 +1,31 @@
+{{/* Common Kafka ENV variables and helpers used by PostHog */}}
+
+{{/* Return the Kafka fullname */}}
+{{- define "posthog.kafka.fullname" -}}
+{{- if .Values.kafka.fullnameOverride -}}
+{{- .Values.kafka.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else if .Values.kafka.nameOverride -}}
+{{- printf "%s-%s" .Release.Name .Values.kafka.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" (include "posthog.fullname" .) "kafka" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+
+{{/* Return the Kafka hosts (brokers) */}}
+{{- define "posthog.kafka.brokers" -}}
+{{- if .Values.kafka.enabled }}
+    {{- printf "%s:%d" (include "posthog.kafka.fullname" .) (.Values.kafka.service.port | int) -}}
+{{- else -}}
+    {{- printf "%s" .Values.externalKafka.brokers -}}
+{{- end -}}
+{{- end -}}
+
+{{/* ENV used by PostHog deployments for connecting to Kafka */}}
+{{- define "snippet.kafka-env" }}
+- name: KAFKA_HOSTS
+  value: "{{ include "posthog.kafka.brokers" . }}"
+- name: KAFKA_URL
+  value: "kafka://{{ include "posthog.kafka.brokers" . }}"
```

Comment:
The format posthog is expecting "kafka://broker0:123,kafka://broker1:123"

This may work assuming the app picks up the first correctly, but I think will not pick up the second: see https://github.com/PostHog/posthog/blob/2e6aae702d3e702188dad0e72e547ed6525ab638/posthog/settings/data_stores.py#L108